modules/openxr/extensions/openxr_eye_gaze_interaction.cpp

```diff
@@ -0,0 +1,98 @@
+/*  openxr_eye_gaze_interaction.cpp                                       */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "openxr_eye_gaze_interaction.h"
+
+#include "core/os/os.h"
+
+#include "../action_map/openxr_interaction_profile_metadata.h"
+
+OpenXREyeGazeInteractionExtension *OpenXREyeGazeInteractionExtension::singleton = nullptr;
+
+OpenXREyeGazeInteractionExtension *OpenXREyeGazeInteractionExtension::get_singleton() {
+	ERR_FAIL_NULL_V(singleton, nullptr);
+	return singleton;
+}
+
+OpenXREyeGazeInteractionExtension::OpenXREyeGazeInteractionExtension() {
+	singleton = this;
+}
+
+OpenXREyeGazeInteractionExtension::~OpenXREyeGazeInteractionExtension() {
+	singleton = nullptr;
+}
+
+HashMap<String, bool *> OpenXREyeGazeInteractionExtension::get_requested_extensions() {
+	HashMap<String, bool *> request_extensions;
+
+	request_extensions[XR_EXT_EYE_GAZE_INTERACTION_EXTENSION_NAME] = &available;
```

Comment:
Was thinking this over some more last night, we need to add a project setting that enables this extension as a whole. We have to keep in mind that what we're doing on Android is not the same as on other platforms.

For most platforms supporting eye gaze, enabling the extension is the trigger that tells the runtime that we intent to use it. As there are privacy concerns here, enabling the eye gaze interaction without actually implementing it could lead to confusion from users.

So this should be an opt in feature changing this code to:
```
if (GLOBAL_GET("xr/openxr/extensions/eye_gaze_interaction")) {
    request_extensions[XR_EXT_EYE_GAZE_INTERACTION_EXTENSION_NAME] = &available;
}
```
And off course adding the appropriate `GLOBAL_DEF` entry.

In the plugin we could either only show the dropdown to select optional/required if this is enabled, and only add the manifest entries if this is enabled.
Or alternatively, if the user changes this setting we could prompt the user whether they want the extension enabled.

Reply:
We should probably do this as a rule with any sensitive extensions that we don't want to auto enable just because they are available.

Reply:
Updated the PR to include the project setting.
The OpenXR loaders PR has also been updating to check for the project setting.

Reply:
I've added it for #81533 as well. We'll have a look at existing extensions and handle those on a case by case basis, hand tracking is indeed a good example. For those however the defaults should be true as to not break existing games.